generic/piranha-tree-sitter/build.rs

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright (c) 2019 Uber Technologies, Inc.
+
+ <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ except in compliance with the License. You may obtain a copy of the License at
+ <p>http://www.apache.org/licenses/LICENSE-2.0
+
+ <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ express or implied. See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+//! Build script. 
+use std::{path::PathBuf, process::{Command, Stdio}};
+
+// Compiles the tree-sitter parser/scanner. 
+// If the tree-sitter source code is not found, it clones the repositories from GitHub 
+// into the folder `piranha/generic/tree-sitter-src` and then generates (and compiles) the parser.c and/or scanner.c
+// Prerequisite: (i) Tree-sitter CLI and (ii) git.
+// Installing tree-sitter's CLI: https://github.com/tree-sitter/tree-sitter/blob/master/cli/README.md
+fn build(language: &str)  -> std::io::Result<()>  {
+    let (ts_src, git_url) = match language {
+        "Java" => ("tree-sitter-java", "https://github.com/tree-sitter/tree-sitter-java.git"),
+        "Swift" => ("tree-sitter-swift","https://github.com/alex-pinkus/tree-sitter-swift.git"),
+        _ => panic!("Language not supported!")
+    };
+    let path_to_all_tree_sitter_src = PathBuf::from(format!("../tree-sitter-src"));
```

Comment:
how about defining a PROJECT_ROOT and specifying this relative to that?

Reply:
Addressed.